python/graphstorm/gpartition/dist_partition_graph.py

```diff
@@ -189,12 +190,33 @@ def main():
             dirs_exist_ok=True,
         )
 
+    # Hard Negative Mapping
+    if args.gsprocessing_config:
+        gsprocessing_config = args.gsprocessing_config
+        shuffle_hard_negative_nids(f"{args.input_path}/{gsprocessing_config}",
+                                   args.num_parts, args.output_path)
+    else:
+        for filename in os.listdir(args.input_path):
+            if filename.endswith("_with_transformations.json"):
+                gsprocessing_config = filename
+                shuffle_hard_negative_nids(f"{args.input_path}/{gsprocessing_config}",
+                                           args.num_parts, args.output_path)
+                break
+        else:
+            # Did not raise error here for not introducing the break change,
+            # but will raise warning here to warn customers.
+            logging.info("Skip the hard negative node ID mapping, "
+                         "please upgrade to the latest GSProcessing.")
+
+
 def parse_args() -> argparse.Namespace:
     """Parses arguments for the script"""
     argparser = argparse.ArgumentParser("Partition DGL graphs for node and edge classification "
                                         + "or regression tasks")
     argparser.add_argument("--input-path", type=str, required=True,
                            help="Path to input DGL chunked data.")
+    argparser.add_argument("--gsprocessing-config", type=str,
```

Comment:
@thvasilo Is there a way to get the path of this config automatically?

Reply:
Good point, if the input data were created by GSProcessing we always save a copy of the gsprocessing config file in the output. We also save the job launch arguments in launch_arguments.json. So if we load `launch_arguments.json` the name of the GSP config file will be available there with the key `config_filename`